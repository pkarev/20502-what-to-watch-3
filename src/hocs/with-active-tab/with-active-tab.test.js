import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withActiveTab from './with-active-tab';

const tabs = [1, 2];

const MockComponent = ({onTabClick}) => (
  <React.Fragment>
    {tabs.map((tab, index) => (
      <div key={index} onClick={() => {
        onTabClick(index);
      }}>
        {tab}
      </div>
    ))}
  </React.Fragment>
);

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  onTabClick: PropTypes.func.isRequired,
};

const MockComponentWithActiveTab = withActiveTab(MockComponent);

it(`Render MockComponent with active tab`, () => {
  const tree = renderer
    .create(
        <MockComponentWithActiveTab>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam, voluptatem?</p>
        </MockComponentWithActiveTab>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
