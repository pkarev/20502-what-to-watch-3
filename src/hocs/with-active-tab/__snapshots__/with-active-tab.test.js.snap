// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MockComponent with active tab 1`] = `
Array [
  <div
    onClick={[Function]}
  >
    1
  </div>,
  <div
    onClick={[Function]}
  >
    2
  </div>,
]
`;
